sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/FileTable.scala

```diff
@@ -34,13 +35,21 @@ import org.apache.spark.sql.util.SchemaUtils
 
 abstract class FileTable(
     sparkSession: SparkSession,
-    options: CaseInsensitiveStringMap,
+    originalOptions: CaseInsensitiveStringMap,
     paths: Seq[String],
     userSpecifiedSchema: Option[StructType])
   extends Table with SupportsRead with SupportsWrite {
 
   import org.apache.spark.sql.connector.catalog.CatalogV2Implicits._
 
+  // Options without path-related options from `originalOptions`.
+  protected final lazy val options: CaseInsensitiveStringMap = {
```

Comment:
Note that this can still masked:
```scala
class DummyFileTable(
    sparkSession: SparkSession,
    options: CaseInsensitiveStringMap, // could be unintentional
    paths: Seq[String],
    expectedDataSchema: StructType,
    userSpecifiedSchema: Option[StructType])
  extends FileTable(sparkSession, options, paths, userSpecifiedSchema) {
```